example/src/App.tsx

```diff
@@ -0,0 +1,106 @@
+import React, { useRef } from 'react'
+
+import {
+  PtsCanvas,
+  HandleStartFn,
+  HandleAnimateFn,
+  HandleActionFn
+} from 'react-pts-canvas'
+import { Pt, Rectangle, Group, Geom, Create, Num, Shaping } from 'pts'
+import './App.css'
+
+const ExampleComponent: React.FC = () => {
+  const gridRef = useRef<Group[] | null>(null)
+  const handleStart: HandleStartFn = (_bound, space) => {
+    if (!space) return
+    gridRef.current = Create.gridCells(
+      space.innerBound,
+      Math.floor(space.width / 50),
+      Math.floor(space.height / 50)
+    )
+  }
+
+  const innerRect = (rectPts: Pt[], t: number, d: number) => {
+    const lines = rectPts.map((r, i) =>
+      i >= 3 ? new Group(r, rectPts[0]) : new Group(r, rectPts[i + 1])
+    )
+    return lines.map(ln =>
+      Geom.interpolate(ln[d > 0 ? 0 : 1], ln[d > 0 ? 1 : 0], t)
+    )
+  }
+
+  const handleAnimate: HandleAnimateFn = (space, form, time) => {
+    if (!form || !space || !gridRef.current || time === undefined) return
```

Comment:
Do we always need to check for undefined here? If I understand correctly, this is due to the optional type declaration. I wonder if there's a better way to make defaults so that we don't need do this check. Thoughts?

Reply:
@tibotiber @williamngan Thats really just a design choice, up to you! By having them be optional and letting users decide how to handle undefined situations lets them write some other side effect if they want to when space or form, etc, are undefined. 

Otherwise, yeah @tibotiber we can just not run handleAnimate if those are undefined, and then users don't have to check. Let me know what way you want to go with that!